File: src/main/scala/org/csvee/CsvConfig.scala
package org.csvee

/** TODO */
final case class CsvConfig(separator: Char, quote: Char, escape: Char) {
  require(
    separator != quote,
    "separator and quote characters must be different"
  )
}

object CsvConfig {
  val default: CsvConfig = CsvConfig(
    separator = ',',
    quote     = '"',
    escape    = '"'
  )
}

File: src/main/java/com/github/lancelet/csvee/CsvConfig.java
package com.github.lancelet.csvee;

public final class CsvConfig {
    public CsvConfig(char separator, char quote, char escape) {
        this.separator = separator;
        this.quote     = quote;
        this.escape    = escape;
    }

    public final char separator;
    public final char quote;
    public final char escape;

    public static final CsvConfig defaults = new CsvConfig(',', '"', '"');
}
